wordpress-mobile/WordPress-iOS
# Magic Email Clients

Master issue Ref #11086

### This PR adds a feature that allows to choose which email client to check for the magic link that we send to login or sign up.

Instead of just opening the mail app, if the user has many email clients, and those fall under the supported email clients, an action sheet will be presented, to choose the email client of preference. 

![IMG_0002b](https://user-images.githubusercontent.com/34376330/73034937-14570b80-3e0c-11ea-8fae-0280ee865758.png)

If no third party client is installed, the existing behavior will still be valid (open the Apple Mail directly)

supported mail clients are listed in info.plist (in the key `LSApplicationQueriesSchemes`)

as well as in `EmailClients.plist`, in `WordPressAuthenticator v. 1.10.7-beta.1`

  - Currently supported email clients:
    - Apple Mail (default)
    - Airmail
    - Fastmail
    - Gmail
    - Microsoft Outlook
    - Spark
    - Yahoo Mail

### Note: it requires `WordPressAuthenticator v. 1.10.7-beta.1`
(related PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/178)

To test:
- `rake dependencies` build, run on a device
- login or signup via magic link
- tap on `open app`:
  - if there are any supported third party client installed (see EmailClients.plist in `WordPressAuthenticator 1.10.7-beta.1`), an action sheet will be presented
  - tap on any button and verify that the related email clients opens
- if there are no third party email clients, verify that `open app` still takes the user to Apple Mail.
PR submission checklist:

  - unit tests were added in WordPressAuthenticator
